File: lib/layer2_core/light/e_surface_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl3;
      case 'standard': return cstl3;
      case 'blue': return cbl3;
      case 'green': return cgl3;
      case 'red': return crl3;
      case 'nothing': return cnol3;
      default: return cstl3;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl3
            AppTheme.STANDARD -> R.color.cstl3
            AppTheme.BLUE -> R.color.cbl3
            AppTheme.GREEN -> R.color.cgl3
            AppTheme.RED -> R.color.crl3
            AppTheme.NOTHING -> R.color.cnol3
        }
    }
}
